### ISSUE TITLE
How to use this action properly? 

### ISSUE THREAD

**Issue by Shaun-Harrison** [mattes/gce-cloudsql-proxy-action#1]
Hi

I've got this action set up within my workflow, I've set the creds and instance, but when the action runs it seems to connect then drop the connection straight away

![image](https://user-images.githubusercontent.com/46562010/98303611-4a8a0400-1fb6-11eb-890a-06d1bc8651b8.png)

I've got a action which runs after this one where I want to drop a database, but as this fails it never gets to it

**Comment by mattes** [comment]
This output looks fine. You're seeing a client establishing and closing a connection to the proxy right after it starts.

**Comment by mattes** [comment]
I see the action fails though (as indicated by the red icon). What does it do after line 72?

**Comment by Shaun-Harrison** [comment]
Error: Process completed with exit code 2.

Thank you!

**Comment by mattes** [comment]
I re-read the code and I think here is what's happening:

1) [It tries to connect 10 times](https://github.com/mattes/gce-cloudsql-proxy-action/blob/325036d428aeddbdbd8bb1d7bc2717320d2a9fb9/action.yml#L49-L60)
2) [It prints the logs](https://github.com/mattes/gce-cloudsql-proxy-action/blob/325036d428aeddbdbd8bb1d7bc2717320d2a9fb9/action.yml#L63)
3) [It exits with code 2](https://github.com/mattes/gce-cloudsql-proxy-action/blob/325036d428aeddbdbd8bb1d7bc2717320d2a9fb9/action.yml#L67)

Exit code 2 indicates that there was no response to the connection attempt.

> pg_isready returns 0 to the shell if the server is accepting connections normally, 1 if the server is rejecting connections (for example during startup), 2 if there was no response to the connection attempt, and 3 if no attempt was made (for example due to invalid parameters).
https://www.postgresql.org/docs/12/app-pg-isready.html

How did you configure your CloudSQL instance? Does it have a public IP? The service account you're using has at least `Cloud SQL Client` role permissions?

**Comment by Shaun-Harrison** [comment]
Hello

It has a public IP, which access is then locked by whitelisting the IP, but obviously this proxy gets around this issue
The Service account has Cloud SQL Admin access 

Here is how I've set it up in the action, perhaps there is an issue with the ordering or something

```
  DATABASE:
    needs: [CHECKS]
    runs-on: ubuntu-latest
    steps:       
    # Reset the database
    - uses: mattes/gce-cloudsql-proxy-action@v1.0.0
      with:
        creds: ${{ secrets.DATABASE_FEATURES_GOOGLE_APPLICATION_CREDENTIALS }}
        instance: $DATABASE_INSTANCE
    - name: drop database
      run: |
        mysql -u $DATABASE_USERNAME -p$DATABASE_PASSWORD -D feature${{ github.event.inputs.feature }}_$DATABASE; -e "DROP DATABASE feature${{ github.event.inputs.feature }}_$DATABASE;"
    - name: cancelling
      if: ${{ failure() }}
      uses: andymckay/cancel-action@0.2
```

**Comment by mattes** [comment]
Oh, it's very possible that this action doesn't work with MySQL right now. I only tested with Postgres. False advertising, sorry. Let me try if I can reproduce with MySQL.

**Comment by Shaun-Harrison** [comment]
That's no problem at all, let me know if you need me to test anything!

**Comment by Piopio** [comment]
Hi,
I've sent a PR for MySQL support (replace pg_isready check by basic TCP check) related to that issue.

**Comment by Piopio** [comment]
@mattes I've forked that repo for the MySQL compatibility.

Will you review my PR (#2 ) or I should keep my fork alive in the future ?

**Comment by mattes** [comment]
Fixed in v1.0.1
